crypto/box/src/traits.rs

```diff
@@ -52,16 +52,24 @@ pub trait CryptoBox<KexAlgo: Kex>: Default {
     ///
     /// Meant to mirror aead::decrypt_in_place_detached
     ///
+    /// NOTE: Meant to run in constant-time even if the mac-check fails.
+    ///
     /// Fails if:
```

Comment:
This feels unclear. For example, "MAC check fails" is listed under both "Fails if:" and "Returns". These two sections could probably be combined into something like:

```
# Returns
* `Ok(true)` if decryption succeeds. `buffer` contains the plaintext and SHOULD be zeroized after use.
* `Ok(false) if MAC check fails. `buffer` contains failed plaintext and SHOULD be zeroized after use.
```
I don't think its helpful to vaguely describe the conditions that cause an Err, e.g. "Anything is wrong with the footer (magic bytes? version code?)" or "error if anything else is wrong". The returned Error type should define and document the possible errors.

Reply:
Fixed.